### ISSUE TITLE
Requirements overhaul

### ISSUE THREAD

**Issue by scotthavens** [USDA-ARS-NWRC/smrf#111]
From `pipdeptree` here are the packages that are install from the `requirements.txt`
```
coloredlogs==6.1
  - humanfriendly [required: >=2.1, installed: 4.6]
coveralls==1.8.2
  - coverage [required: >=3.6,<5.0, installed: 4.5.4]
  - docopt [required: >=0.6.1, installed: 0.6.2]
  - requests [required: >=1.0.0, installed: 2.20.0]
    - certifi [required: >=2017.4.17, installed: 2017.11.5]
    - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
    - idna [required: >=2.5,<2.8, installed: 2.7]
    - urllib3 [required: >=1.21.1,<1.25, installed: 1.23]
Cython==0.28.4
flake8==3.5.0
  - mccabe [required: >=0.6.0,<0.7.0, installed: 0.6.1]
  - pycodestyle [required: >=2.0.0,<2.4.0, installed: 2.3.1]
  - pyflakes [required: >=1.5.0,<1.7.0, installed: 1.5.0]
inicheck==0.3.5
  - pandas [required: >=0.18.1, installed: 0.22.0]
    - numpy [required: >=1.9.0, installed: 1.14.0]
    - python-dateutil [required: >=2, installed: 2.6.1]
      - six [required: >=1.5, installed: 1.11.0]
    - pytz [required: >=2011k, installed: 2017.3]
mock==2.0.0
  - pbr [required: >=0.11, installed: 3.1.1]
  - six [required: >=1.9, installed: 1.11.0]
mysql-connector-python-rf==2.2.2
netCDF4==1.2.9
  - numpy [required: >=1.7, installed: 1.14.0]
pipdeptree==0.13.2
  - pip [required: >=6.0.0, installed: 19.2.3]
progressbar2==3.6.2
  - argparse [required: Any, installed: 1.1]
PyKrige==1.4.0
  - matplotlib [required: >=2.0.0, installed: 3.1.1]
    - cycler [required: >=0.10, installed: 0.10.0]
      - six [required: Any, installed: 1.11.0]
    - kiwisolver [required: >=1.0.1, installed: 1.1.0]
      - setuptools [required: Any, installed: 41.2.0]
    - numpy [required: >=1.11, installed: 1.14.0]
    - pyparsing [required: >=2.0.1,!=2.1.6,!=2.1.2,!=2.0.4, installed: 2.4.2]
    - python-dateutil [required: >=2.1, installed: 2.6.1]
      - six [required: >=1.5, installed: 1.11.0]
  - numpy [required: >=1.14.0, installed: 1.14.0]
  - scipy [required: >=1.0.0, installed: 1.0.0]
    - numpy [required: >=1.8.2, installed: 1.14.0]
spatialnc==0.2.10
sphinx-rtd-theme==0.2.5b1
sphinxcontrib-bibtex==0.3.5
  - latexcodec [required: >=0.3.0, installed: 1.0.5]
    - six [required: >=1.4.1, installed: 1.11.0]
  - oset [required: >=0.1.3, installed: 0.1.3]
    - setuptools [required: Any, installed: 41.2.0]
  - pybtex [required: >=0.17, installed: 0.21]
    - latexcodec [required: >=1.0.4, installed: 1.0.5]
      - six [required: >=1.4.1, installed: 1.11.0]
    - PyYAML [required: >=3.01, installed: 4.2b1]
  - pybtex-docutils [required: >=0.2.0, installed: 0.2.1]
    - docutils [required: >=0.8, installed: 0.14]
    - pybtex [required: >=0.16, installed: 0.21]
      - latexcodec [required: >=1.0.4, installed: 1.0.5]
        - six [required: >=1.4.1, installed: 1.11.0]
      - PyYAML [required: >=3.01, installed: 4.2b1]
    - six [required: Any, installed: 1.11.0]
  - six [required: >=1.4.1, installed: 1.11.0]
  - Sphinx [required: >=1.0, installed: 1.6.6]
    - alabaster [required: >=0.7,<0.8, installed: 0.7.10]
    - babel [required: >=1.3,!=2.0, installed: 2.5.1]
      - pytz [required: >=0a, installed: 2017.3]
    - docutils [required: >=0.11, installed: 0.14]
    - imagesize [required: Any, installed: 0.7.1]
    - Jinja2 [required: >=2.3, installed: 2.10.1]
      - MarkupSafe [required: >=0.23, installed: 1.0]
    - Pygments [required: >=2.0, installed: 2.2.0]
    - requests [required: >=2.0.0, installed: 2.20.0]
      - certifi [required: >=2017.4.17, installed: 2017.11.5]
      - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
      - idna [required: >=2.5,<2.8, installed: 2.7]
      - urllib3 [required: >=1.21.1,<1.25, installed: 1.23]
    - setuptools [required: Any, installed: 41.2.0]
    - six [required: >=1.5, installed: 1.11.0]
    - snowballstemmer [required: >=1.1, installed: 1.2.1]
    - sphinxcontrib-websupport [required: Any, installed: 1.0.1]
utm==0.4.0
wheel==0.33.6
```

Packages that are solely for develop and testing purposes
- coveralls
- flake8
- Sphinx
- sphinx-rtd-theme
- sphinxcontrib-bibtex
- sphinxcontrib-websupport

Not needed:
- progressbar2
- setuptools
- wheel

The requirements.txt can be split into two different files, a base version and one for development and testing.

**Comment by micahjohnson150** [comment]
I think this is merged in and complete. I think we can just close it right?

**Comment by scotthavens** [comment]
Yes, it was merged in PR #112
